content/en/blog/2024/otel-operator-q-and-a/index.md

```diff
@@ -0,0 +1,340 @@
+[OTel Demo’s Docker compose file](https://github.com/open-telemetry/opentelemetry-demo/blob/06f020c97f78ae9625d3a4a5d1107c55045c567f/docker-compose.yml#L665-L668).
+
+Unfortunately, while the OTel Collector supports multiple Collector
+configuration files, the Collector managed by the OTel Operator does not.
+
+To get around this, you could merge the multiple Collector configs through some
+external tool beforehand. For example, if you
+[were deploying the Operator via Helm](https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-operator),
+you could technically
+[pass it multiple Collector config files using multiple --values flags](https://stackoverflow.com/a/56653384)
+and let [Helm](https://helm.sh) do the merging for you.
+
+For reference,
+[check out this thread in the #otel-operator CNCF Slack channel](https://cloud-native.slack.com/archives/C033BJ8BASU/p1709321896612279).
+
+### Q2: How can I securely reference access tokens in the OpenTelemetryCollector’s configuration?
+
+In order to send OpenTelemetry data to an observability backend, you must define
+at least one [exporter](/docs/collector/configuration/#exporters). Whether you
+use [OTLP](/docs/specs/otel/protocol/) or
+[some proprietary vendor format](/docs/specs/otel/protocol/), most exporters
+typically require that you specify an endpoint and an access token when sending
+data to a vendor backend.
+
+When using the OpenTelemetry Operator to manage the OTel Collector, the OTel
+Collector config YAML is defined in the
+[OpenTelemetryCollector](https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#getting-started)
+CR. This file should be version-controlled and therefore shouldn’t contain any
+sensitive data, including access tokens stored as plain text.
+
+Fortunately, the `OpenTelemetryCollector` CR gives us a way to reference that
+value as a secret. Here’s how you do it:
+
+1- Create a Kubernetes secret for your access token. Remember to
+[base-64 encode](https://www.base64encode.org/) the secret.
+
+2-
+[Expose the secret as an environment variable](https://kubernetes.io/docs/concepts/configuration/secret/#using-a-secret)
+by adding it to the `OpenTelemetryCollector` CR’s
+[`env` section](https://github.com/avillela/otel-target-allocator-talk/blob/21e9643e28165e39bd79f3beec7f2b1f989d87e9/src/resources/02-otel-collector-ls.yml#L16-L21).
+For example:
+
+```yaml
+env:
+  - name: LS_TOKEN
+    valueFrom:
+      secretKeyRef:
+        key: LS_TOKEN
+        name: otel-collector-secret
+```
+
+3- Reference the environment variable in your
+[exporter definition](https://github.com/avillela/otel-target-allocator-talk/blob/21e9643e28165e39bd79f3beec7f2b1f989d87e9/src/resources/02-otel-collector-ls.yml#L43-L47):
+
+```yaml
+exporters:
+  otlp/ls:
+    endpoint: 'ingest.lightstep.com:443'
+    headers:
+      'lightstep-access-token': '${LS_TOKEN}'
+```
+
+For more info, check out my full example
```

Comment:
Maybe we should also mention authentication option in the exporter?  
Not sure if its worth to mention, but in case you use some auth container that writes such token into a file, there is the bearer token auth ext. that can be used to watch the file and set the header.

Reply:
@frzifus TBH I don't know that I would be qualified to talk about this, just because I've never played around with the authentication option in the exporter.